src/coreclr/interpreter/compiler.cpp

```diff
@@ -1543,7 +1550,21 @@ void InterpCompiler::getEHinfo(CORINFO_METHOD_INFO* methodInfo, unsigned int ind
                 ehClause->TryOffset = 0;
                 ehClause->TryLength = (uint32_t)m_ILCodeSizeFromILHeader;
                 ehClause->HandlerOffset = (uint32_t)m_synchronizedFinallyStartOffset;
-                ehClause->HandlerLength = (uint32_t)(m_synchronizedPostFinallyOffset - m_synchronizedFinallyStartOffset);
+                ehClause->HandlerLength = (uint32_t)(m_synchronizedOrAsyncPostFinallyOffset - m_synchronizedFinallyStartOffset);
+                ehClause->ClassToken = 0;
+                return;
+            }
+        }
+        else if (m_isAsyncMethodWithContextSaveRestore)
+        {
+            // Logically we append the synchronized finally clause at the end of the EH clauses, as it holds all of the methodbody within its try region, and nested eh clauses are required to be before their containing clauses.
```

Comment:
A nit - a copy and paste issue - comment talks about synchronized finally clause, but this one is for async finally.